pysmt/simplifier.py

```diff
@@ -364,8 +354,12 @@ def walk_times(self, formula, args, **kwargs):
         new_args = sorted(new_args, key=FNode.node_id)
         return self.manager.Times(new_args)
 
-
     def walk_pow(self, formula, args, **kwargs):
+        if args[0].is_algebraic_constant():
+            from pysmt.constants import Numeral
+            l = args[0].constant_value()
+            r = args[1].constant_value()
+            return self.manager._Algebraic(Numeral(l**r))
```

Comment:
Again, a simple test showing that this has the semantics you expect would be good.
Could you move this as the last case of this if? I expect this to be false more often than not.

Reply:
I have added a simple test `test_times_algebraic` that is supposed to test this.